src/Http/Routing/src/ShortCircuit/RouteShortCircuitEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing;
+
+/// <summary>
+/// 
+/// </summary>
+public static class RouteShortCircuitEndpointRouteBuilderExtensions
+{
+    private static readonly RequestDelegate _shortCircuitDelegate = (context) => Task.CompletedTask;
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="statusCode"></param>
+    /// <param name="routePrefixes"></param>
+    /// <returns></returns>
+    public static IEndpointConventionBuilder MapShortCircuit(this IEndpointRouteBuilder builder, int statusCode, params string[] routePrefixes)
+    {
+        var group = builder.MapGroup("");
+        foreach (var routePrefix in routePrefixes)
+        {
+            string route;
+            if (routePrefix.EndsWith("/", StringComparison.OrdinalIgnoreCase))
+            {
+                route = $"{routePrefix}{{**catchall}}";
+            }
+            else
+            {
+                route = $"{routePrefix}/{{**catchall}}";
+            }
+            group.Map(routePrefix, _shortCircuitDelegate);
+        }
+
+        return group.ShortCircuit(statusCode);
+    }
+}
```

Comment:
How does this compose with the existing `MapFallback*` APIs?

Reply:
Why use MapGroup at all?

Reply:
> How does this compose with the existing MapFallback* APIs?

Could you please elaborate the problem a little bit. I'm not that much familiar with `MapFallback` APIs. Does it set the endpoint if there's no matching Endpoint? in which case `MapShortCircuit` with only "/" as a prefix would be the problem, right?

> Why use MapGroup at all?

I wanted to return one `IEndpointConventionBuilder` which has all the endpoints and `MapGroup` seems like the easiest way to have this functionality. Should I write a new class implementing `IEndpointConventionBuilder`?

Reply:
I agree groups are convenient for this. The one problem I see is that even though the declared type is `IEndpointConventionBuilder`, someone could call `(RouteGroupBuilder)app.MapShortCircuit(404, "/prefix1", "/prfix2")` and then add additional routes.

While not exactly a problem, it's a little weird, and would prevent us from returning another `IEndpointConventionBuilder` that's not derived from the currently-sealed `RouteGroupBuilder` without going through the breaking change process since it'd be considered a [behavioral breaking change](https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules#behavioral-changes) to the return value. So probably better to be safe and copy the private [CompositeEndpointConventionBuilder](https://github.com/dotnet/aspnetcore/blob/33e7a9a111742c8b06924383baba5c448d7f8af8/src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs#L131). Maybe we could make it shared source.

Reply:
That's a good point for MapGroup. As long as it doesn't cause any path match conflicts when you pass in "".

Reply:
@halter73 @Tratcher what's the final decision here? Keeping the mapGroup or implementing IEndpoinConventionBuilder?

Reply:
Offline advice from @halter73:

> [MapGroup is] not terrible. I'd probably just add the metadata to each endpoint though and use a private aggregate IEndpointConventionBuiler implementation to prevent people from casting it back to a RouteGroupBuilder though

Reply:
@Kahbazi there are a group of APIs that map wildcard segments already with very low priority, this will override that.

For example, if you call this API, it'll potentially break Blazor apps and SPAs because the new routes are very general and do not have an order set.

I think these APIs should be fine-tuned for the scenarios they are trying to target or leave to the application to do so.

Reply:
@javiercn I tried this with fallback and other wildcard routes and it worked fine. The routeprefix was sufficient to distinguish.